packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -827,6 +860,123 @@ export class ResourceService {
     return userAction;
   }
 
+  private async validateNewResourcesData(
+    newServices: RedesignProjectNewService[],
+    project: Project
+  ): Promise<void> {
+    const projectResources = await this.prisma.resource.findMany({
+      where: {
+        projectId: project.id,
+        resourceType: EnumResourceType.Service,
+        deletedAt: null,
+      },
+    });
+    if (projectResources.length > 0) {
+      for (const newService of newServices) {
+        // duplicate name validation
+        const duplicateService = projectResources.find(
+          (resource) =>
+            resource.name.toLocaleLowerCase() ===
+            newService.name.toLocaleLowerCase()
+        );
+        if (duplicateService) {
+          throw new AmplicationError(
+            `Resource : ${newService.name} is already exist in project: ${project.name}.`
+          );
+        }
+        //service limitation validation
+        const featureServices = await this.billingService.getMeteredEntitlement(
+          project.workspaceId,
+          BillingFeature.Services
+        );
+
+        if (
+          !featureServices.hasAccess ||
+          (!featureServices.isUnlimited &&
+            featureServices.usageLimit <= projectResources.length)
```

Comment:
If multiple services are added, you are only checking if one can be added (same validation inside the for loop for each service, not taking into account newServices.length)